### PR TITLE
Bytsa/2244 info about provider when pressing the f5 button when editing information about the provider the page does not return to its previous state

### PR DESCRIPTION
**Pull request by Ultrafiolet4**
#2244

### CODE REVIEW

**Comment by humanmantis** [comment]
Have you tried this? https://www.ngxs.io/plugins/storage
We can store a part of our state in SessionStorage, and it allows us to keep everything under control in the NGXS State

**Comment by Ultrafiolet4** [comment]
No, I haven't. I think it doesn't matter here, we have the same sessionStorage by default.

**Comment by doliinyk** [comment]
Fixed provider edit info page refresh issue & used NGXS Storage Plugin

File: src/app/shared/components/provider-info/provider-info.component.ts [hunk 1]
```diff
@@ -40,17 +41,15 @@ export class ProviderInfoComponent implements OnInit, OnDestroy {
   public institutionStatusName: string;
   public destroy$: Subject<boolean> = new Subject<boolean>();
 
-  constructor(private store: Store) {}
+  public constructor(private store: Store) {}
```
Comment (humanmantis) [inline src/app/shared/components/provider-info/provider-info.component.ts]: Remove access modifier for constructor

File: src/app/shell/personal-cabinet/shared-cabinet/create-form/create-form.component.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export abstract class CreateFormComponent implements OnInit, OnDestroy {
   public isPristine = true;
   public editMode: boolean;
 
-  constructor(protected store: Store, protected route: ActivatedRoute, protected navigationBarService: NavigationBarService) {}
+  protected constructor(protected store: Store, protected route: ActivatedRoute, protected navigationBarService: NavigationBarService) {}
```
Comment (humanmantis) [inline src/app/shell/personal-cabinet/shared-cabinet/create-form/create-form.component.ts]: Remove access modifier for constructor

File: src/app/shell/personal-cabinet/provider/create-provider/create-provider.component.ts [hunk 3]
```diff
@@ -50,9 +42,13 @@ import { CreateFormComponent } from '../../shared-cabinet/create-form/create-for
 export class CreateProviderComponent extends CreateFormComponent implements OnInit, AfterViewInit, OnDestroy, AfterViewChecked {
   @ViewChild('stepper') public stepper: MatStepper;
 
+  @Select(RegistrationState.provider)
+  private provider$: Observable<Provider>;
+
   public provider: Provider;
   public isAgreed: boolean;
   public isNotRobot: boolean;
+  public isEditMode: boolean = this.store.selectSnapshot(AppState.isEditMode);
```
Comment (humanmantis) [inline src/app/shell/personal-cabinet/provider/create-provider/create-provider.component.ts]: Move selecting snapshot to ngOnInit

**Review by humanmantis — CHANGES_REQUESTED** [review]

**Review by humanmantis — APPROVED** [review]

**Pull request merged**